File: 2.Agentes de resolução de Problemas/1.Modelos/python/state.py
from transition import Transition
class State:
    def __init__(self, name: str, edges = []):
        self.name = name
        self.edges = edges

    def __eq__(self, __o) :
        return self.name == __o.name

    def copy(self):
        res = State(self.name)
        res.edges = self.edges.copy()
        return res
    
    def getEdges(self):
        return [ [e.target,e.cost] for e in self.edges]
    def show(self):
        s = 'name: {0}\n'.format(self.name)
        s+= 'edge:\n'
        for e in self.edges:
            s+="\t({0},{1})\n".format(e.target,e.cost)
        print(s)

File: 2.Agentes de resolução de Problemas/1.Modelos/java/src/Map/State.java
package Map;

import java.util.ArrayList;
import java.util.List;

public class State {
    public String name;
    public List<Transition> edges;
    public State father;
    public State(String name, List<Transition> edges){
        this.name = name;
        this.edges = edges;
    }
    public State(){
        this.name = "";
        this.edges = new ArrayList<Transition>();
    }

    public String show(){
        var s = String.format("%s:\n", this.name);
        for (Transition transition : edges) {
            s+=transition.show();
        }
        return s;
    }
}
